spec/acceptance/reporting_spec.rb

```diff
@@ -55,54 +55,83 @@
     master.idempotent_apply(setup_manifest)
   end
 
-  context 'with report status: unchanged' do
-    context 'and noop_pending: false' do
-      let(:sitepp_content) do
-        # Puppet should report that nothing happens
-        ''
+  context 'with default incident_creation_report_attributes' do
+    context 'with report status: unchanged' do
+      context 'and no pending changes' do
+        let(:sitepp_content) do
+          # Puppet should report that nothing happens
+          ''
+        end
+
+        include_examples 'no incident'
       end
 
-      include_context 'reporting test setup'
+      context 'and pending changes' do
+        let(:sitepp_content) do
+          to_manifest(declare('notify', 'foo', 'noop' => true))
+        end
+
+        include_context 'incident creation test setup'
 
-      it 'does nothing' do
-        num_incidents_before_puppet_run = IncidentHelpers.get_incidents('').length
-        trigger_puppet_run(master, acceptable_exit_codes: [0])
-        num_incidents_after_puppet_run = IncidentHelpers.get_incidents('').length
-        expect(num_incidents_after_puppet_run).to eql(num_incidents_before_puppet_run)
+        include_examples 'no incident'
       end
     end
 
-    context 'and noop_pending: true' do
+    context 'with report status: intentional changes' do
+      # Default parameters report on corrective changes, not intentional changes.
       let(:sitepp_content) do
-        to_manifest(declare('notify', 'foo', 'noop' => true))
+        to_manifest(declare('notify', 'foo'))
       end
 
       include_context 'incident creation test setup'
-      include_examples 'incident creation test', 'noop_pending'
+      include_examples 'no incident'
     end
-  end
 
-  context 'with report status: changed' do
-    let(:sitepp_content) do
-      to_manifest(declare('notify', 'foo'))
+    context 'with report status: corrective changes' do
+      include_context 'incident creation test setup'
+      include_context 'corrective change'
+      include_examples 'incident creation test', 'changed'
     end
 
-    include_context 'incident creation test setup'
-    include_examples 'incident creation test', 'changed'
+    context 'with report status: failed changes' do
+      let(:sitepp_content) do
+        to_manifest(declare('exec', 'foo', 'command' => '/bin/foo_command'))
+      end
+
+      include_context 'incident creation test setup'
+      include_examples 'incident creation test', 'failed'
+    end
   end
 
-  context 'with report status: failed' do
-    let(:sitepp_content) do
-      to_manifest(declare('exec', 'foo', 'command' => '/bin/foo_command'))
+  context 'with user-specified incident_creation_report_statuses' do
```

Comment:
This would be a good place to test the negative of the default tests. So
```
  // set incident_creation_report_attributes = [all_attributes] - [defaults]
  context "report has 'intentional_changes' attribute" do
    // incident creation test
  end
  context "report has 'unchanged' attribute" do
    // incident creation test
  end
  ...
  context "report does not have one of the user-specified attributes" do
    context "report has 'failed_changes' attribute" do
      // incident noop test
    end
    ...
  end
```

(Notice the [complement] symmetry with the default test cases).

Note there's also typo -- `report_statuses` => `report_attributes`.

Reply:
You already have several of these tests btw, just making them a bit more complete. Main addition's the extra noop tests. My thinking there is it'd be nice to test that e.g. a report with a 'failed_changes' attribute _can_ noop on incident creation if it's not included in the incident creation attributes-- we can test that condition via the user-specified attributes.

Reply:
Resolving b/c these aren't PR blockers